processor/transformer/transformer.go

```diff
@@ -270,9 +270,19 @@ func (trans *handle) transform(
 	if len(clientEvents) == 0 {
 		return Response{}
 	}
-
 	sTags := statsTags(&clientEvents[0])
 
+	var trackWg sync.WaitGroup
+	defer trackWg.Wait()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	trackWg.Add(1)
+	go func() {
+		trackLongRunningTransformation(ctx, stage, sTags, trans.config.timeoutDuration, trans.logger)
```

Comment:
```suggestion
		trackLongRunningTransformation(ctx, stage, trans.config.timeoutDuration, trans.logger.With(sTags))
```

Reply:
`With` accepts varargs, whereas `sTags` is a `map[string]string`